app/src/main/java/com/example/homework1calculator/Calculator.kt

```diff
@@ -1,6 +1,189 @@
 package com.example.homework1calculator
 
 
+class Calculator {
+
+    private val regFindNum = Regex("[-*/+]")
+    private val regFindSigns = Regex("[^-+*/=]")  //(?<=[^-+*/=,{ ])[-+*/=]
+    private val regexSpace = Regex("[\\s]")
+    private val priorityMap = mapOf(
+
+        "+" to 1,
+        "-" to 1,
+        "*" to 2,
+        "/" to 2
+    )
+
+    fun calculate(text: String?): String {
+
+        var finalResult = ""
+
+        if (text != null) {
+            if (text.contains('(')) {
+                finalResult = calcWithBraces(text)
+
+            } else finalResult = calculateWithNoBraces(text)
+
+        }
+
+        return finalResult
+
+    }
```

Comment:
из if можно вернуть результат + форматирование никакое, ну и вопрос, если ты не предусматриваешь null строку может быть проще сделать так `fun calculate(text: String): String`
из-за твоего решения у тебя каждая функция теперь принимает null
```suggestion
    fun calculate(text: String?): String {
        return text?.let { 
            if (it.contains('(')) {
                calcWithBraces(it)
            } else { 
                calculateWithNoBraces(it)
            }
        } ?: ""
    }
```